common/djangoapps/student/views/management.py

```diff
@@ -910,7 +911,11 @@ def confirm_email_change(request, key):
 
         response = render_to_response("email_change_successful.html", address_context)
 
-        USER_EMAIL_CHANGED.send(sender=None, user=user)
+        if settings.ENFORCE_SESSION_EMAIL_MATCH:
+            # Update the JWT cookies with new user email
+            response = update_jwt_cookies_in_response(request, response, user)
```

Comment:
Would it be possible for this to be a part of the `USER_EMAIL_CHANGED` receiver code so we know any email change would get this? Would it be too late for the middleware to take care of this in process_response if register_email_change was called? Something like that? I haven't thought this through, but hopefully you can.

Reply:
I marked a variable in register change and moved this logic in `process_response` of the middleware if that variable is marked. Could you please give your opinion on this?

Reply:
Thanks. I'm going to add comments on that code.